### ISSUE TITLE
Loki Output - remove_keys skipped if used in labels

### ISSUE THREAD

**Issue by DevSusu** [fluent/fluent-bit#5546]
## Bug Report

**Describe the bug**
Using Loki output, `remove_keys` don't work as expected when values within `remove_keys` are used by `labels`

using `fluent/fluent-bit:1.9.4`

**To Reproduce**

```yaml
# docker-compose.yaml
version: "3.7"

services:
  fluent-bit:
    image: fluent/fluent-bit
    network_mode: "host"
    volumes:
      - ./fluent-bit.yaml:/fluent-bit/etc/fluent-bit.yaml
    command: "/fluent-bit/bin/fluent-bit -c /fluent-bit/etc/fluent-bit.yaml"
```

```yaml
# fluent-bit.yaml
env:
  log_level: debug
  flush_interval: 5

pipeline:
  inputs:
  - dummy:
      dummy: '{"log":"blahblah","kubernetes":{"pod_name":"nginx", "namespace_name":"default"}}'

  outputs:
  - loki:
      match: '*'
      auto_kubernetes_labels: off
      labels: pod=$kubernetes['pod_name']
      remove_keys: kubernetes
```

```bash
$ docker-compose up

$ nc -l 3100
POST /loki/api/v1/push HTTP/1.1
Host: 127.0.0.1:3100
Content-Length: 148
User-Agent: Fluent-Bit
Content-Type: application/json

{"streams":[{"stream":{"pod":"nginx"},"values":[["1654686265055834768","{\"log\":\"blahblah\",\"kubernetes\":{\"namespace_name\":\"default\"}}"]]}]}
```

- Steps to reproduce the problem:

**Expected behavior**
```
# expected
{"streams":[{"stream":{"pod":"nginx"},"values":[["1654686265055834768","{\"log\":\"blahblah\"}"]]}]}

# actual
{"streams":[{"stream":{"pod":"nginx"},"values":[["1654686265055834768","{\"log\":\"blahblah\",\"kubernetes\":{\"namespace_name\":\"default\"}}"]]}]}
```

**Your Environment**
* Version used: 1.9.4
* Configuration: given above
* Environment name and version: docker
* Server type and version:
* Operating System and version: ubuntu 20.04
* Filters and plugins: Loki

**Additional context**
using `tail` input and `kubernetes` filter and `loki` output in kubernetes environment, and ran into this problem. There are a lot of useless labels and annotations included in logs. I want only relevant information and remove key `kubernetes`

**Comment by DevSusu** [comment]
adding more context.

when not referencing `kubernetes` from labels, keys are removed as configured

```yaml
# fluent-bit.yaml
env:
  log_level: debug
  flush_interval: 5

pipeline:
  inputs:
  - dummy:
      dummy: '{"log":"blahblah","kubernetes":{"pod_name":"nginx", "namespace_name":"default"}}'

  outputs:
  - loki:
      match: '*'
      auto_kubernetes_labels: off
      remove_keys: kubernetes
```

```bash
$ docker-compose up

$ nc -l 3100
POST /loki/api/v1/push HTTP/1.1
Host: 127.0.0.1:3100
Content-Length: 105
User-Agent: Fluent-Bit
Content-Type: application/json

{"streams":[{"stream":{"job":"fluent-bit"},"values":[["1654685781055956259","{\"log\":\"blahblah\"}"]]}]}
```

please help me out identifying if this is intentional, or a bug

**Comment by nandha4083** [comment]
I am also facing the same issue. Key "kubernetes" specified in remove_keys does not have any effect. I am still seeing it in stream output. 

I am using few other keys like "stream", "filename" etc in labels and when specify it in "remove_keys" it has an effect and I don't see those keys in output stream which is expected.

Fluent-bit version: 1.9.4

Sample Output plugin:
```
    [Output]
        Name loki
        Match *
        host xxxx
        port 3100
        tenant_id ""
        labels  filename=$filename, stream=$stream, namespace=$kubernetes['namespace_name']
        remove_keys stream,filename,kubernetes
        auto_kubernetes_labels false
        line_format json
```

**Comment by nokute78** [comment]
Thank you for reporting. I think it is a bug.

out_loki tries to remove key/value using `flb_mp_accessor_keys_remove`.
The API needs a key lists to indicate which key should be removed.

I tested the API and I found that the list should be sorted if it is passed nested key/value.
However the lists is not sorted at current master. It may not work if user passes nested key/value.
https://github.com/fluent/fluent-bit/pull/5590/files#r901076390

I pushed a test code to detect this issue.
https://github.com/fluent/fluent-bit/pull/5590/commits/8e6f2eb01b4b6d475dc6abec7a87b33dc28a4c19
We need a patch to sort the list.

**Comment by DevSusu** [comment]
@nokute78 thank you for working on this! Hope to see test & code fixes soon :smile:

**Comment by easoncon** [comment]
When will this feature be updated, I really need this feature，thank you。

**Comment by mericano1** [comment]
Any update on this ? There seems to be an MR open

**Comment by DevSusu** [comment]
@nokute78 thanks a lot for working on this

**Comment by braunsonm** [comment]
Still an issue. Thanks @nokute78 for working on it!

**Comment by GrzegorzWild** [comment]
Also waiting for this. Currently we use promtail and fluentbit daemonsets, and this bug fix would allow us to use just one log agent (fluentbit).

**Comment by endersonmaia** [comment]
I have a similar situation, and can confirm it's happening in `fluent-bit:2.0.6`.

**Comment by endersonmaia** [comment]
I've updated to 2.0.10 and it's still happening. :cry:

`remove_keys` only works if I don't use `kubernetes` in the `labels`

**Comment by nokute78** [comment]
The patch is not merged to v2.0.x.
Could you try v2.1.x+ ?
